packages/gatsby-source-contentful/src/extend-node-type.js

```diff
@@ -215,7 +219,9 @@ const resolveFluid = (image, options) => {
   fluidSizes = fluidSizes.map(Math.round)
 
   // Filter out sizes larger than the image's maxWidth.
-  const filteredSizes = fluidSizes.filter(size => size <= width)
+  const filteredSizes = fluidSizes.filter(
+    size => size <= contentful_max_size || size <= width
```

Comment:
This line has the same issue as above but there is a catch 😂

A few lines below we have the following line of code
https://github.com/gatsbyjs/gatsby/blob/ca4f15af9a68302a78436af60801e973b9907416/packages/gatsby-source-contentful/src/extend-node-type.js#L228

This one fights with the contentful_max_size constant. If the image is bigger than contentful_max_size we still add it.

So I suggest changing that line to
`if (!filteredSizes.includes(parseInt(width)) && parseInt(width) < contentful_max_size) filteredSizes.push(width)`

Reply:
aah nice!!